clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java

```diff
@@ -108,54 +61,25 @@ public InputStream download(Artifact artifact) throws IOException {
     Path stagingPath =
         Paths.get(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString());
 
-    if (!isValidReference(repoReference)) {
-      throw new IOException(
-          "Artifact reference "
-              + repoReference
-              + " is invalid for artifact account with auth type "
-              + authType);
-    }
+    executor.clone(repoReference, remoteRef, stagingPath);
+    String repoBasename = getRepoBasename(repoReference);
+    Path outputFile = Paths.get(stagingPath.toString(), repoBasename + ".tgz");
 
-    try (Closeable ignored = () -> FileUtils.deleteDirectory(stagingPath.toFile())) {
```

Comment:
@kevinawoo called this out already but this try-with-resource is used to ensure that the staging directory is cleaned up properly after the repo has been cloned. You still want this unless you've accounted for that elsewhere and I missed it.

Reply:
Nice finding! I totally missed that